lib/unhangout-sockets.js

```diff
@@ -243,6 +320,30 @@ _.extend(UnhangoutSocketManager.prototype, events.EventEmitter.prototype, {
             }
         }, this));
 
+        mgr.on("participant-proposed", _.bind(function(socket, args) {
+            var event = this.getEvent(args.roomId);
+            
+            if (event && socket.user && socket.user.isAdminOf(event)) {
+                event.save({adminProposedSessions: false});
+                mgr.writeAck(socket, "participant-proposed");
+                event.logAnalytics({action: "participant-proposed", user: socket.user});
+            } else {
+                mgr.writeErr(socket, "participant-proposed");
+            }
+        }, this));
+
```

Comment:
Consider combining "participant-proposed" and "admin-proposed" into one, with an argument to set participant proposed to "true" or "false".  This will reduce code duplication.